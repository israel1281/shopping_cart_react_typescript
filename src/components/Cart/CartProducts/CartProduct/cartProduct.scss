.cartProductContainer {
  position: relative;
  box-sizing: border-box;
  padding: 5%;
  transition: background-color 0.2s, opacity 0.2s;

  &::before {
    content: "";
    width: 90%;
    height: 2px;
    background-color: rgba(255, 255, 255, 0.678);
    position: absolute;
    top: 0;
    left: 5%;
  }

  .cartDeleteButton {
    top: 15px;
    right: 5%;
    position: absolute;
    color: #fff;
    border: none;
    font-size: 1rem;
    z-index: 2;
    cursor: pointer;
    border: 0;
    background-color: transparent;

    &:focus-visible {
      outline: 3px solid white;
    }

    &:hover {
      background-position-x: -17px;
    }
  }

  .cartImage {
    display: inline-block;
    vertical-align: middle;
    width: 15%;
    height: auto;
    margin-right: 3%;
  }

  .cartDetail {
    width: 57%;
    display: inline-block;
    vertical-align: middle;

    .cartDetailTitle {
      color: #ececec;
      margin: 0;
    }
    .cartDetailDesc {
      color: #ffffff;
      margin: 0;
    }
  }

  .cartPrice {
    display: inline-block;
    vertical-align: middle;
    color: #fff;
    text-align: right;
    width: 25%;

    .changeQuantity {
      color: rgb(0, 44, 65);
      border: 0;
      background-color: rgba(242, 171, 6, 0.938);
      width: 25px;
      height: 25px;
      font-weight: 600;
      cursor: pointer;
      margin-top: 0.4rem;

      &:focus-visible {
        outline: 3px solid black;
      }

      &:disabled {
        opacity: 0.2;
      }
    }
  }
}
