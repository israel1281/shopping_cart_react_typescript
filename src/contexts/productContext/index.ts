export { ProductsProvider } from "./ProductContextProvider";
export { default as useProducts } from "./useProducts";
