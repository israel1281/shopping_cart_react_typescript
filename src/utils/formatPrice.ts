const formatPrice = (price: number, currencyId: string): string => {
  switch (currencyId) {
    case "NGN":
      return price.toFixed(4).replace(".", ",");
    default:
      return price.toFixed(2);
  }
};

export default formatPrice;
