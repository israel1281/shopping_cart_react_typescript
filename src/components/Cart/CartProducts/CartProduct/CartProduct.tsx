import formatPrice from "../../../../utils/formatPrice";
import { ICartProduct } from "../../../../models";

import { useCart } from "../../../../contexts/cartContext";
import { MdCancel } from "react-icons/md";
import "./cartProduct.scss";

interface IProps {
  product: ICartProduct;
}
const CartProduct = ({ product }: IProps) => {
  const { removeProduct, increaseProductQuantity, decreaseProductQuantity } =
    useCart();
  const {
    sku,
    title,
    price,
    style,
    currencyId,
    currencyFormat,
    availableSizes,
    quantity,
  } = product;

  const handleRemoveProduct = () => removeProduct(product);
  const handleIncreaseProductQuantity = () => increaseProductQuantity(product);
  const handleDecreaseProductQuantity = () => decreaseProductQuantity(product);

  return (
    <div className="cartProductContainer">
      <button onClick={handleRemoveProduct} className="cartDeleteButton">
        <MdCancel />
      </button>

      <img
        src={require(`../../../../assets/products/${sku}-1-cart.webp`)}
        alt={title}
        className="cartImage"
      />
      <div className="cartDetail">
        <p className="cartDetailTitle">{title}</p>
        <p className="cartDetailDesc">
          {`Size: ${availableSizes[0]} | Color: ${style}`} <br />
          Quantity: {quantity}
        </p>
      </div>
      <div className="cartPrice">
        <p>{`${currencyFormat}  ${formatPrice(price, currencyId)}`}</p>
        <div>
          <button
            onClick={handleDecreaseProductQuantity}
            disabled={quantity === 1 ? true : false}
            className="changeQuantity"
          >
            -
          </button>
          <button
            onClick={handleIncreaseProductQuantity}
            className="changeQuantity"
          >
            +
          </button>
        </div>
      </div>
    </div>
  );
};

export default CartProduct;
