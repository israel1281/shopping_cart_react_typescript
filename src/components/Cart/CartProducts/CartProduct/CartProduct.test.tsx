import { CartProvider } from "../../../../contexts/cartContext";
import { mockCartProducts } from "../../../../utils/mockData";

import CartProduct from ".";

describe("[CartProduct Component", () => {
  const setup = (props = {}) => {
    return (
      <CartProvider>
        <CartProduct product={mockCartProducts[0]} {...props} />
      </CartProvider>
    );
  };

  test("should render correctly", () => {
    const view = setup();
    expect(view).toMatchSnapshot();
  });
});
