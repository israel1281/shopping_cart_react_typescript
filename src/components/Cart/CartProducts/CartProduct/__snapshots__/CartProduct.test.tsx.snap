// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CartProduct Component should render correctly 1`] = `
<CartProvider>
  <CartProduct
    product={
      Object {
        "availableSizes": Array [
          "S",
          "L",
          "XL",
          "XXL",
        ],
        "currencyFormat": "$",
        "currencyId": "USD",
        "description": "14/15 s/nÂº",
        "id": 0,
        "installments": 9,
        "isFreeShipping": true,
        "price": 10.9,
        "quantity": 1,
        "sku": 8552515751438644,
        "style": "White T-shirt",
        "title": "Cropped Stay Groovy off white",
      }
    }
  />
</CartProvider>
`;
